logs_sdk/lib/opentelemetry/sdk/logs/read_write_log_record.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Logs
+      # A {LogRecord} interface that can write to the full {LogRecord} and
+      # retrieve all information added to the {LogRecord}.
+      class ReadWriteLogRecord < OpenTelemetry::Logs::LogRecord
+        attr_accessor :timestamp,
+                      :observed_timestamp,
+                      :trace_id,
+                      :span_id,
+                      :trace_flags,
+                      :severity_text,
+                      :severity_number,
+                      :body,
+                      :attributes,
+                      :resource,
+                      :information_scope
+
+        # Creates a new {ReadWriteLogRecord}.
+        #
+        # @param [optional Float, Time] timestamp Time when the event occurred.
+        # @param [optional String] trace_id The trace ID associated with a {ReadWriteLogRecord}.
+        # @param [optional String] span_id The span ID associated with a {ReadWriteLogRecord}.
+        # @param [optional TraceFlags] trace_flags The trace flags associated with a {ReadWriteLogRecord}.
+        # @param [optional String] severity_text The log severity, also known as log level.
+        # @param [optional Integer] severity_number The numerical value of the log severity.
+        # @param [optional String, Numeric, Boolean, Array<String, Numeric,
+        #   Boolean>, Hash{String => String, Numeric, Boolean, Array<String,
+        #   Numeric, Boolean>}] body The body of the {ReadWriteLogRecord}.
+        # @param [optional Hash] attributes Additional information about the event.
+        #
+        # @return [ReadWriteLogRecord]
+        def initialize(
+          timestamp: nil,
+          trace_id: nil,
+          span_id: nil,
+          trace_flags: nil,
+          severity_text: nil,
+          severity_number: nil,
+          body: nil,
+          attributes: nil,
+          logger: nil
```

Comment:
* [JS uses the logger to get to the instrumentation scope and resource](https://github.com/open-telemetry/opentelemetry-js/blob/b400c2e5d9729c3528482781a93393602dc6dc9f/experimental/packages/sdk-logs/src/LogRecord.ts#L100-L101)
* [I think PHP uses the logger for the resource, but not for the instrumentation scope](https://github.com/open-telemetry/opentelemetry-php/blob/62a14d1c20db456baabbabf003f5efdbf0b030da/src/SDK/Logs/ReadableLogRecord.php#L51-L59)
* [Java seems to set them based on what's passed in](https://github.com/open-telemetry/opentelemetry-java/blob/90ab3665dc0fb56598fcf9b9c15941eedb3822f8/sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkReadWriteLogRecord.java#L51-L52)
* Python [(approx location)](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py#L150) seems to accept args for resource and gets the scope through the LogData object

I'm tempted to follow JS here and pass in the logger to get those attributes, but I'm open to other ideas.